docs/reference-guides/data/data-core-block-editor.md

```diff
@@ -846,6 +846,19 @@ _Returns_
 
 -   `string|false`: Block Template Lock
 
+### hasAllowedPatterns
+
+Returns whether there is at least one allowed pattern for inner blocks children. This is useful for deferring the parsing of all patterns until needed.
+
+_Parameters_
+
+-   _state_ `Object`: Editor state.
+-   _rootClientId_ `?string`: Optional target root client ID.
```

Comment:
The root client ID is not really optional, it just defaults to `null` (the whole document) when not specified.